### ISSUE TITLE
Repository idea: Add an operations runbook for query-embedding cache behavior and disk-budget tuning

### ISSUE THREAD

**Issue by jeanibarz** [jeanibarz/knowledge-base-mcp-server#711]
# Repository Idea Recommendation: Add an operations runbook for query-embedding cache behavior and disk-budget tuning

## Summary
Category: documentation. Turn the existing ADR/config surface for query-embedding cache into an operator-facing runbook: what cache statuses mean, how disk budget pruning behaves, and when to bypass or clear it.

## Project context
The project is a local-first knowledge-base MCP server and `kb` CLI with strong retrieval, indexing, local LLM, diagnostics, transport, and test surfaces. The relevant existing features are documented in `features.md` for this run.

## Current feature evidence
`src/query-cache.ts` implements L1/disk cache, corruption counters, disk limit enforcement. `docs/feature-flags.md` lists cache flags. `docs/architecture/adr/0009-query-embedding-cache.md` records design rationale.

## Existing capability check
Positive evidence: Design and config details exist.

Negative evidence: There is no operator-facing `docs/operations/query-cache.md` style guide.

## Existing issue search
The corrected open issue snapshot contains zero open issues. Closed issue and PR searches were run for this candidate using the query matrix in `duplicate-evidence.md`. Adjacent work is listed below and was used only as context, not treated as novelty proof.

## Duplicate evidence table
| Candidate | Query | Surface | Top matching issue/PR URLs | Duplicate risk | Distinction |
| --- | --- | --- | --- | --- | --- |
| Add an operations runbook for query-embedding cache behavior and disk-budget tuning | `"query embedding cache" operations runbook` | issue/pr | https://github.com/jeanibarz/knowledge-base-mcp-server/issues/427 | low | No exact same outcome found in all-state issue and PR search; adjacent results are implementation/background only. |
| Add an operations runbook for query-embedding cache behavior and disk-budget tuning | `"KB_QUERY_CACHE" documentation` | issue/pr | none | low | No exact same outcome found in all-state issue and PR search; adjacent results are implementation/background only. |
| Add an operations runbook for query-embedding cache behavior and disk-budget tuning | `"query-cache" disk budget` | issue/pr | https://github.com/jeanibarz/knowledge-base-mcp-server/issues/491, https://github.com/jeanibarz/knowledge-base-mcp-server/issues/214, https://github.com/jeanibarz/knowledge-base-mcp-server/issues/203 | low | No exact same outcome found in all-state issue and PR search; adjacent results are implementation/background only. |
| Add an operations runbook for query-embedding cache behavior and disk-budget tuning | `"hit_l1" "hit_disk" docs` | issue/pr | https://github.com/jeanibarz/knowledge-base-mcp-server/issues/646, https://github.com/jeanibarz/knowledge-base-mcp-server/issues/216, https://github.com/jeanibarz/knowledge-base-mcp-server/issues/214 | low | No exact same outcome found in all-state issue and PR search; adjacent results are implementation/background only. |
| Add an operations runbook for query-embedding cache behavior and disk-budget tuning | `query embedding cache pruning tuning` | issue/pr | none | low | No exact same outcome found in all-state issue and PR search; adjacent results are implementation/background only. |

## Knowledge base grounding
# KB Evidence

- backend-systems/backend-systems/notes/2605.20706.md lines 7-14

  Excerpt: Architecture topic cloud-native  ## Design pattern The system uses a static memory arena for all GPU allocations, avoiding dynamic buffer creation and fragmentation. A tunable kernel library separates runtime scheduling from shader generation, enabling device-specific optimization (e.g., tile sizes, subgroup operations) while maintaining portability. Quantization-aware kernels dequantize weights on-the-fly into shared memory or registers during matrix operations, supporting 23 formats without pe

- backend-systems/backend-systems/notes/2605.11517.md lines 4-8

  Excerpt: Summary This paper addresses the memory capacity bottleneck in full-graph Graph Neural Network (GNN) training, which typically requires massive GPU memory or expensive distributed clusters. The authors introduce GriNNder, a framework that leverages NVMe SSD storage to offload intermediate activations and gradients, enabling single-GPU training for graphs that previously required multi-server setups. The system employs Structured Storage Offloading (SSO), coordinating cache, regather, and bypass 

Stale/prefix warning observed:
2026-06-19T22:12:52.181Z [INFO] Applying embedding task prefixes for nomic-embed-text:latest (documents: "search_document: ", queries: "search_query: ") — indexes built without prefixes (pre-#567) need a reindex

## Recommended idea
Operators see query-cache statuses in CLI output and config flags, but the behavior is scattered across code, ADRs, and feature flags instead of a runbook explaining hit_l1/hit_disk/miss/bypass, disk budget pruning, and safe cache clearing.

Implement: Write a runbook with cache lifecycle, env vars, telemetry, troubleshooting, clearing/pruning guidance, and examples for interpreting search JSON/canonical logs.

## Why this is not a duplicate
No open upstream issue requests this same outcome. Closed adjacent issues and merged PRs cover neighboring surfaces or prerequisites, but not this exact capability. The duplicate-search artifact records the matching URLs and distinctions.

## Why this angle
Turn the existing ADR/config surface for query-embedding cache into an operator-facing runbook: what cache statuses mean, how disk budget pruning behaves, and when to bypass or clear it.

## Minimal implementation or validation path
Write a runbook with cache lifecycle, env vars, telemetry, troubleshooting, clearing/pruning guidance, and examples for interpreting search JSON/canonical logs.

Validation: Add doc anchor checks and examples that match the existing CLI JSON contract.

## Risks and open questions
Keep the first version narrow. Preserve existing CLI/MCP contracts, avoid exposing secrets or raw KB content by default, and document any compatibility behavior. The main open question is the exact command/flag name where this touches CLI UX.

## Files and issues inspected
`src/query-cache.ts`, `docs/feature-flags.md`, ADR 0009, issues #214 and #491.

---
Generated by the Repository Idea Scout playbook.
State: `/home/jean/.kookr/playbook-state/repository-idea-scout/jeanibarz-knowledge-base-mcp-server/2e85129e-579b-4b05-a732-e12e53af83e4/recommendations/03-query-cache-operations-runbook`
